unit/util/symbol_table.cpp

```diff
@@ -17,14 +17,55 @@ TEST_CASE("Iterating through a symbol table", "[core][utils][symbol_tablet]")
 
   symbol_table.insert(symbol);
 
-  int counter = 0;
-  for(auto &entry : symbol_table)
+  SECTION("Non const iterator")
```

Comment:
This commit (a9afb9a25f3fd32e5e227bc9a91d7d5228bf38f6) can also be removed; symbol table iterators are tested in `unit/util/symbol_table.cpp`

Reply:
I can see a single class to `begin()` but no call to end. This also tests the correct dispatching when you have a `symbol_tablet` but the compile time type is `symbol_table_baset`